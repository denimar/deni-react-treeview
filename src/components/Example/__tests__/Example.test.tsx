import React from 'react';
import { Example } from '..';
import { render } from '@testing-library/react';

describe('Example Component', () => {
  it('Example renders', () => {
    const { container } = render(<Example />);

    const node = container.querySelector('button.Example');
    expect(node!.className).toEqual('Example');
  });

  it('Matches snapshot', () => {
    const { container } = render(<Example />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Example active renders', () => {
    const { container } = render(<Example active />);

    const node = container.querySelector('button.Example');
    expect(node!.className).toEqual('Example Example--active');
  });

  it('Example disabled renders', () => {
    const { container } = render(<Example disabled />);

    const node = container.querySelector('button.Example');
    expect(node!.className).toEqual('Example Example--disabled');
  });

  it('Example children renders', () => {
    const { container } = render(<Example>foo</Example>);

    const node = container.querySelector('button.Example');
    expect(node!.innerHTML).toEqual('foo');
  });

  it('Example extra classNames render', () => {
    const { container } = render(<Example className="foo" />);

    const node = container.querySelector('button.Example');
    expect(node!.className).toEqual('Example foo');
  });
});
