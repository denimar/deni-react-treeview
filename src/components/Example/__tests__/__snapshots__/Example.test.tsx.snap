// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Component Matches snapshot 1`] = `
<button
  class="Example"
/>
`;
